### DISCUSSION TITLE
[Feature/Help Request] adding the ability to change the calendar type.

### DISCUSSION THREAD

**Discussion by hisothreed** [henninghall/react-native-date-picker#430] [Ideas]
Hello, So basically I'm trying to add the option to change the calendar type (for example from Gregorian to Hijri/islamic calendar), so using the locale ID alone obviously is not possible, and so as a major enhancement for this library I would suggest adding the option to change the calendar type, I've started doing that locally by modifying the package's code and basically got it working on iOS by adding this to ```RNDatePickerManager.m```:

```
RCT_CUSTOM_VIEW_PROPERTY(calendar, NSString, DatePicker)
{
    NSCalendarIdentifier calendarType = NSCalendarIdentifierGregorian;
    NSString* type = json;
    if ([type  isEqual: @"islamic"]) {
        calendarType = NSCalendarIdentifierIslamic;
    }
    if ([type  isEqual: @"gregorian"]) {
        calendarType = NSCalendarIdentifierGregorian;
    }
    view.calendar = [NSCalendar calendarWithIdentifier:calendarType];
}

```

trying to do the same for android but I'm not familiar with android development overall and the library's structure, so if anyone please can point out where and what should I change on android.
for now I want to fork and change the code for my self first then I will try to clean it up and submit a new PR to add it as a feature (by then this will be the only library supporting different types of calendars), your help and support would be highly appreciated.

Thank you.

**Comment by HoseinAsadi** [comment]
+1
